src/chart.ts

```diff
@@ -241,7 +241,7 @@ export class Chart extends LitElement {
           const { state } = normalizeStateValue(
             this.config.unit_prefix,
             Number(subEntity.state),
-            subEntity.attributes.unit_of_measurement || unit_of_measurement,
+            this._getUnitOfMeasurement(entityConf.unit_of_measurement || subEntity.attributes.unit_of_measurement)

```

Comment:
I don't think this is right. For charts that don't show energy with date selection it will use the wrong unit. For date selection I don't see where unit conversion happens in `energy.ts`. The only conversion I see is with `convert_units_to` but that is already handled in `this._getUnitOfMeasurement`. Does HA return converted values from `recorder/statistics_during_period`? I haven't seen that

Reply:
The conversion happens in `energy.ts` at the end of `getStatistics()` with `result[id] *= scale`, where `scale` is set based on `convert_units_to`.

`_getUnitOfMeasurement()` just makes sure that `normalizeStateValue()` doesn't try to *1000 due to the unit being `kWh`, since `kWh` was already converted to `gCO2` in `getStatistics()`

Reply:
> For charts that don't show energy with date selection it will use the wrong unit

I don't see what here depends on energy with date selection, but I do think this will be wrong if the main entity has an config-overrided unit that has a different prefix than that of the sub-entity. Really that `entityConf` should be `subEntityConf`. I was being lazy and neglecting that possibility. I can fix that.